packages/nextjs/src/app-router/client/ClerkProvider.tsx

```diff
@@ -52,8 +49,8 @@ export const useNextRouter = (): ClerkHostRouter => {
     shallowPush(path: string) {
       canUseWindowHistoryAPIs ? window.history.pushState(null, '', path) : router.push(path, {});
     },
-    pathname: () => pathname,
-    searchParams: () => searchParams,
+    pathname: () => window.location.pathname,
+    searchParams: () => new URLSearchParams(window.location.search),
```

Comment:
Can you please provide some more context about this change?

Reply:
I had the same question

Reply:
@nikosdouvlis @panteliselef sure! With the way we are passing the router into clerk-js, these values become stale as we are closing over the values at the time of initial render in the provider here. The only way to get these values from Next is through the hooks, so we would need to update this router object on every render and ensure the clerk components re-render based on that.

As our components only render on the client, the source of truth _should_ be the URL, so it seems safe to do that here. This way it ensures the value is always fresh.

Reply:
We've historically had problems with trying to use the `window.location` instead of what the router believes is currently the location.

The challenge is that each router implements its own method of changing pages. It's an async process that, at a minimum, requires `fetch()`ing the next page and rerendering.

The exact timing of when `window.history.pushState` gets called is an implementation detail, so looking at changes there isn't a tremendously reliable signal for the page having changed.  The inconsistency in implementations led to us creating an `awaitableNavigate()` in various SDKs.

All this is to say - I agree the source of _should_ be the URL, and to some extent it is, but the lack of precision means we should at least be very cautious here. Was there anything motivating this change from a functionality perspective, or was it changed optimistically as a cleaner approach?

Reply:
> but the lack of precision means we should at least be very cautious here

For the reasons Colin stated above, I'd prefer we test the change in isolation on a separate PR. I'd like to merge this today to unblock us from running the e2e using the new components and then spend some time verifying this works.

Reply:
Guys, this change only impacts the router used by elements and the rebuilt components, can we agree to merge and do further testing?